WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift

```diff
@@ -65,21 +65,30 @@ struct UpgradesView: View {
                         Task {
                             await upgradesViewModel.purchasePlan(with: plan.wpComPlan.id)
                         }
+                    }, onDismiss: {
+                        upgradesViewModel.trackDismiss(step: .planDetails)
```

Comment:
This happens too often. I suspect it's on the wrong `onDismiss` modifier (because OwnerUpgradesView is a component of the screen which gets swapped out as the state changes, it's not the screen itself), or `onDismiss` doesn't work as we'd hope it would.

Ideally, we only want this to be tracked when the view actually slides off screen... currently it happens when the plan feature benefits are opened, when the waiting state starts, and possibly other times too.